### PR TITLE
feat: show inline error message on username page

### PR DESCRIPTION
**Pull request by barelyhuman**
adds api to check if the user exists and also stop navigation if it doesn't

Closes: #14

### CODE REVIEW

**Comment by danielroe** [comment]
what about immediate navigating, showing skeleton loader, then showing toast if there's a 404 from the commit API, and then navigating _back_ to home page if so?

what I'm worried about is that the fetching of the username info will take as long as otherwise loading the card would - and it won't give an immediate indication that the page is loading

**Comment by barelyhuman** [comment]
Understood, we could switch to using the ungh api for user data, it’ll be faster if the delay is reason. 

If that’s equally slow then i guess the solution would be to do the double redirect

**Comment by danielroe** [comment]
I'm suggesting we don't need a second API request when navigating, but just display toast + redirect(?) after the first one returns 404.

Though I guess we could also try an as-you-type validation on the username text box?

File: app.vue [hunk 1]
```diff
@@ -37,3 +37,14 @@ useServerHead({
     </footer>
   </div>
 </template>
+
+
+<!-- global styles -->
+<style>
+
+.toastify.error{
```
Comment (danielroe) [inline app.vue:45]: Should we remove this now?

File: package.json [hunk 2]
```diff
@@ -18,10 +18,12 @@
     "nuxt": "3.7.0",
     "nuxt-og-image": "^2.0.23",
     "nuxt-time": "^0.1.1",
+    "toastify-js": "^1.12.0",
```
Comment (danielroe) [inline package.json:21]: I think we can remove this

File: pages/[username].vue [hunk 3]
```diff
@@ -184,15 +196,18 @@ const { data: commit, error } = await useFetch(`/api/commit/${username}`, {
   lazy: true
 })
 
-if (process.server && !commit.value) {
-  throw createError({ statusCode: 404 })
-}
+// if (process.server && !commit.value) {
```
Comment (danielroe) [inline pages/[username].vue:199]: Should we remove this commented-out code?

File: pages/[username].vue [hunk 4]
```diff
@@ -219,8 +234,17 @@ const message = computed(() => {
...
+})
 
+const errorMessage = computed(()=>{
+  if(error){
+    return error.value?.data.message || ""
+  }
+  return null
 })
+
+console.log({errorMessage:errorMessage.value});
```
Comment (danielroe) [inline pages/[username].vue:246]:
```suggestion
```

File: tailwind.config.ts [hunk 5]
```diff
@@ -1,5 +1,6 @@
 import { Config } from 'tailwindcss'
 export default {
+  content: ["./lib/**/*.ts"],
```
Comment (danielroe) [inline tailwind.config.ts:3]: I think we can remove this now

File: server/api/commit/[username].get.ts [hunk 6]
```diff
@@ -34,11 +34,7 @@ export default defineEventHandler(async event => {
   }
   const commit = ResultsSchema._parse(results.value).output?.items[0]
   if (!commit) {
-    // @ts-expect-error unknown
-    if (results?.total_count) {
-      console.log(JSON.stringify(ResultsSchema._parse(results).issues))
-    }
-    throw createError({ statusCode: 404, message: 'no commits found' })
+    throw createError({ statusCode: 404, message: 'no commits to show' })
```
Comment (danielroe) [inline server/api/commit/[username].get.ts:37]:
```suggestion
    // @ts-expect-error unknown
    if (results?.total_count) {
      console.log(JSON.stringify(ResultsSchema._parse(results).issues))
    }
    throw createError({ statusCode: 404, message: 'no commits to show' })
```

**Review by danielroe — APPROVED** [review]
beautiful - thank you!

File: pages/[username].vue [hunk 7]
```diff
@@ -219,8 +221,15 @@ const message = computed(() => {
   }
 
   return `Check out ${username}'s first commit on GitHub.`
+})
 
+const errorMessage = computed(()=>{
+  if(error){
+    return error.value?.data.message || ""
+  }
```
Comment (danielroe) [inline pages/[username].vue:229]:
```suggestion
  if (error) {
    return error.value?.data.message || ''
  }
```

**Pull request merged**